src/edu/csus/ecs/pc2/core/Constants.java

```diff
@@ -191,18 +191,70 @@ private Constants() {
     public static final int KIBIBYTE_PER_MEBIBYTE = 1024;
     
     /**
-     * Sandbox constants
+     * Directory where scripts go that PC2 executes for sandbox and interactive submissions.
+     * This folder is in the PC2 home folder (install folder).
+     */
+    public static final String PC2_SCRIPT_DIRECTORY = "scripts";
+    
+    /**
+     * Command line to run submission in a sandbox
      */
     public static final String PC2_INTERNAL_SANDBOX_COMMAND_LINE = "./{:sandboxprogramname} {:memlimit} {:timelimit}";
 
+    /**
+     * File name of script to run submission in sandbox ({:sandboxprogramname} if non-interactive)
+     */
     public static final String PC2_INTERNAL_SANDBOX_PROGRAM_NAME = "pc2sandbox.sh";
     
+    /**
+     * Filename of the script to run submission for an interactive problem in a sandbox ({:sandboxprogramname} if interactive)
+     */
+    public static final String PC2_INTERNAL_SANDBOX_INTERACTIVE_NAME = "pc2sandbox_interactive.sh";
+    
+    /**
+     *  Command to execute to run an interactive problem in a sandbox
+     */
+    public static final String PC2_INTERNAL_SANDBOX_INTERACTIVE_COMMAND_LINE = "./{:sandboxprogramname} {:memlimit} {:timelimit} {:validator} {:infilename} {:ansfilename} {:testcase}";
+    
+    /**
+     *  Filename of the script to run for an interactive problem when not using a sandbox
+     */
+    public static final String PC2_INTERACTIVE_NAME = "pc2_interactive.sh";
+    
+    /**
+     *  Command to execute to run an interactive problem when not using a sandbox
+     */
+    public static final String PC2_INTERACTIVE_COMMAND_LINE = "./pc2_interactive.sh {:validator} {:infilename} {:ansfilename} {:testcase}";
+    
+    /**
+     *  Filename of the script to run during the validate phase AFTER the interactive validator has completed
+     */
+    public static final String PC2_INTERACTIVE_VALIDATOR_NAME = "pc2validate_interactive.sh";
+    
+    /**
+     *  Command to execute to validate results of an interactive run AFTER the interactive validator has completed
+     */
+    public static final String PC2_INTERACIVE_VALIDATE_COMMAND = "./pc2validate_interactive.sh {:resfile} {:feedbackdir} {:testcase}";
```

Comment:
Since `PC2_INTERACTIVE_VALIDATOR_NAME` is a predefined constant (see above), the constant definition should be used in this line rather than the hard-coded string _pc2validate_interactive.sh_.

Reply:
Good point.  Will be fixed in next commit.